### ISSUE TITLE
Get meta data of currently logged in user.

### ISSUE THREAD

**Issue by aprilmintacpineda** [auth0/react-native-auth0#348]
### Describe the problem

I've been looking at the documentation but I can't find any information about it. Every user in the system has metadata, in that metadata are some data that I need, here's the flow:

1. User logs in (using web login, `auth0.webAuth.authorize`)
2. If login successful, get user metadata.

Here's how I'm trying to do it, based on google search results directing me to some issues in this repo, and also the Github's search bar, like this one https://github.com/auth0/react-native-auth0/issues/197

```js
const credentials = await auth0.webAuth.authorize({
  scope: 'openid profile email read:current_user',
  connection
});

const { accessToken } = credentials;
const { sub } = await auth0.auth.userInfo({ token: accessToken });
const user = await auth0.users({ token: accessToken }).getUser({ id: sub });
```

I would get:

```bash
Bad Request: Bad HTTP authentication header format
    at responseHandler (http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false:263790:11)
    at http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false:263818:18
    at http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false:143005:24
    at flush (http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false:143688:11)
    at tryCallOne (http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false:3509:14)
    at http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false:3610:17
    at http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false:30747:21
    at _callTimer (http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false:30635:9)
    at _callImmediatesPass (http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false:30674:7)
    at MessageQueue.callImmediates [as _immediatesCallback] (http://192.168.1.4:8081/index.bundle?platform=ios&dev=true&minify=false:30891:14)
```

### Reproduction

Please read the description above.

### Environment

- **Version of this library used:** 2.6.0
- **Which framework are you using, if applicable:**
- **Other modules/plugins/libraries that might be involved:**
- **Any other relevant information you think would be useful:**

**Comment by Widcket** [comment]
@aprilmintacpineda the ID Token (present in the credentials) should contain the same data as the `userInfo` response. You only need to decode it, as the library [already validates it](https://github.com/auth0/react-native-auth0/blob/master/src/jwt/validator.js). Does that solve your issue?

**Comment by aprilmintacpineda** [comment]
@Widcket No. What I need to get is the user meta data of the user, the userInfo doesn't have that.

![image](https://user-images.githubusercontent.com/21032419/103245782-39e58100-499c-11eb-9fad-0c7829f02bb4.png)

**Comment by Widcket** [comment]
@aprilmintacpineda the method `getUser` belongs to the [Management API](https://auth0.com/docs/api/management/v2#!/Users/get_users_by_id), and you need an AT for that API. Check out [this documentation](https://auth0.com/docs/microsites/manage-users/define-maintain-custom-user-data#customize-and-maintain-user-data) about the different ways to access metadata, eg. using a Rule to copy it into the ID Token

**Comment by Widcket** [comment]
Closing due to inactivity.

**Comment by aprilmintacpineda** [comment]
> @aprilmintacpineda the method `getUser` belongs to the [Management API](https://auth0.com/docs/api/management/v2#!/Users/get_users_by_id), and you need an AT for that API. Check out [this documentation](https://auth0.com/docs/microsites/manage-users/define-maintain-custom-user-data#customize-and-maintain-user-data) about the different ways to access metadata, eg. using a Rule to copy it into the ID Token

@Widcket sorry I haven't really been able to check this, I might check it out next week when the tasks are active again, but essentially what I want to achieve here is get the user's meta data when the user logs in (we are using the web login), at first I checked if it's on the tokens provided but it seems it isn't, so I wondered if there's an endpoint that I need to hit or a method that I need to call to get the user's meta data.

So basically the solution is, it's part of the `idToken` but there are configurations to be done on the auth0 side to make that happen, is that right?

**Comment by Widcket** [comment]
There are essentially 2 ways of accessing the user's metadata:

1. Make a request from your app to the [Management API](https://auth0.com/docs/users/manage-users-using-the-management-api). You'll need to request an AT for that API (by including the Management's API identifier in the audience array -> `https://YOUR_DOMAIN/api/v2/`) and include the following scope `read:current_user`.
2. Use a Rule to add the metadata directly onto the ID Token.
> there are configurations to be done on the auth0 side to make that happen, is that right?

Exactly, please refer to [this docs page](https://auth0.com/docs/rules/metadata).

**Comment by aprilmintacpineda** [comment]
Thank you very much for the help, we will try the `idToken` approach :bow:
